File: FutterlisteNg.Domain/Services/IUserService.fs
namespace FutterlisteNg.Domain.Services
    
open FutterlisteNg.Domain.Model
open FutterlisteNg.Domain.Data
    
type IUserService =
    abstract member All : seq<User>
    
type UserService(repository : IUserRepository) =
    interface IUserService with
        member this.All = repository.All

File: Domain/Service/IUserService.cs
using System.Collections.Generic;
using System.Threading.Tasks;
using FutterlisteNg.Data.Model;

namespace FutterlisteNg.Domain.Service
{
    public interface IUserService
    {
        Task<IEnumerable<User>> FindAllAsync();
        Task<User> GetAsync(string username);
        Task AddAsync(User toAdd);
        Task DeleteAsync(string username);
        Task UpdateAsync(User toUpdate);
    }
}